pkg/search/postgres/store.go

```diff
@@ -135,3 +142,41 @@ func (s *GenericStore[T, PT]) Count(ctx context.Context) (int, error) {
 
 	return RunCountRequestForSchema(ctx, s.schema, sacQueryFilter, s.db)
 }
+
+// Walk iterates over all the objects in the store and applies the closure.
+func (s *GenericStore[T, PT]) Walk(ctx context.Context, fn func(obj PT) error) error {
+	var sacQueryFilter *v1.Query
+	if s.hasPermissionsChecker() {
+		if ok, err := s.permissionChecker.WalkAllowed(ctx); err != nil {
+			return err
+		} else if !ok {
+			return sac.ErrResourceAccessDenied
```

Comment:
This changes the behaviour compared to the generated code.
I'd stick to the previous behaviour that returns nil, otherwise, this may leak information about the user allowed scope.